var/spack/repos/builtin/packages/llvm-openmp-ompt/package.py

```diff
@@ -33,26 +33,69 @@ class LlvmOpenmpOmpt(CMakePackage):
 
     homepage = "https://github.com/OpenMPToolsInterface/LLVM-openmp"
 
-    # tr4-stable branch
+    # towards_tr4 branch
+    version('towards_tr4', branch='towards_tr4',
+            git='https://github.com/OpenMPToolsInterface/LLVM-openmp.git')
+
     version('3.9.2b2',
             git='https://github.com/OpenMPToolsInterface/LLVM-openmp.git',
             commit='5cdca5dd3c0c336d42a335ca7cff622e270c9d47')
+
     # align-to-tr-rebased branch
     version('3.9.2b',
             git='https://github.com/OpenMPToolsInterface/LLVM-openmp.git',
             commit='982a08bcf3df9fb5afc04ac3bada47f19cc4e3d3')
 
+    # variant for building llvm-openmp-ompt as a stand alone library
+    variant('standalone', default=False,
+            description="Build llvm openmpi ompt library as a \
+                         stand alone entity.")
+
+    # variant for building llvm-openmp-ompt using the toward_tr4 branch
+    variant('towardstr4', default=False,
+            description="Use towards_tr4 branch of llvm openmpi \
+                         ompt library for build.")
+
+    # variant for building llvm-openmp-ompt using the gnu compilers
+    variant('usegnu', default=False,
```

Comment:
Same here, u can ask spec for its compiler and do if-else

Reply:
Do you mean something like this?
 if 'gcc' in self.compiler.cc:
                cmake_args.extend(
                    ['-DCMAKE_C_COMPILER=gcc',
                     '-DCMAKE_CXX_COMPILER=g++'])
  else:
                cmake_args.extend(
                    ['-DCMAKE_C_COMPILER=clang',
                     '-DCMAKE_CXX_COMPILER=clang++'])

Reply:
not exactly, `if self.spec.satisfies('%gcc')` or alike.